ug: ts: add common aklite reports (#533)
foundriesio/docs: source/user-guide/troubleshooting/troubleshooting.rst

Errors and Solutions
---------------------

Aktualizr-Lite Common Reports
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

This section shows some common returns from ``aktualizr-lite`` operations.

.. tip::
   To get the ``aktualizr-lite`` logs, run::

      sudo journalctl -fu aktualizr-lite

* **curl error 3**

.. code-block::

   curl error 3 (http code 0): URL using bad/illegal format or missing URL

> This could mean that the device is not properly registered, but there is a dummy ``/var/sota/sql.db`` file available. This file can be created in case ``aktualizr-lite`` is manually run before registering the device.

Reviewer: ```suggestion
This could mean that the device is not properly registered, but there is a dummy ``/var/sota/sql.db`` file available. This file may be created when ``aktualizr-lite`` is manually run before registering the device.
```
Unsure about context, suggestion rewords to say this dummy file may be created when aktualizr-lite is run before registering

**Solution:** Re-flash or :ref:`ref-ts-re-register`. Make sure the device is properly registered before running ``aktualizr-lite``.

* **curl error 6 or 56**

.. code-block::

   curl error 6 (http code 0): Couldn't resolve host name

Or:

.. code-block::

   curl error 56 (http code 0): Failure when receiving data from the peer

> These could mean that there is no networking available and/or the device cannot talk to device gateway. This could be due to a bad registration or a dummy ``/var/sota/sql.db`` file. These can also mean that the DNS is not working correctly (for instance IPv6 only).

Reviewer 2: ```suggestion
These could mean that there is no networking available and/or the device cannot talk to the device gateway. This could be due to a lousy registration or a dummy ``/var/sota/sql.db`` file. These can also mean that the DNS is not working correctly (for instance, IPv6 only).
```
This sentence starts with "these could mean" and after there is a "this could be". I'm a little bit confused on how to review the plural on this, as I think the first "these" is related to both error 6 and 56. Could you double check?

Author: "This could be" should refer to "the device cannot talk to the device gateway".

Author: Any idea how to make this clearer?

> **Solution:** Re-flash or :ref:`ref-ts-re-register`. If you are :ref:`Setting up your Device Gateway PKI <ref-rm-pki>`, make sure all operations have succeed.

Reviewer 3: s/have succeed./have succeeded./g

* **curl error 7 or 28**

.. code-block::

   curl error 7 (http code 0): Couldn't connect to server

Or:

.. code-block::

   curl error 28 (http code 0): Timeout was reached

These could mean that a device cannot reach the server.

> **Solution:** Make sure your device has a good connection. Check for proxy or firewalls in the network. If you are :ref:`Setting up your Device Gateway PKI <ref-rm-pki>`, make sure all operations have succeed.

Reviewer 3: s/have succeed./have succeeded./g

> **Solution:** Make sure your device has a good connection. Check for proxy or firewalls in the network. If you are :ref:`Setting up your Device Gateway PKI <ref-rm-pki>`, make sure all operations have succeeded.

Reviewer 2: ```suggestion
**Solution:** Make sure your device has a good connection. Check for proxies or firewalls in the network. If you are :ref:`Setting up your Device Gateway PKI <ref-rm-pki>`, make sure all operations have succeeded.
```

* **Failed to update Image repo metadata**

.. code-block::

   Failed to update Image repo metadata: The root metadata was expired.

This means your TUF root key has expired.

**Solution:** Rotate your :ref:`ref-offline-keys`.

.. code-block::

   Failed to update Image repo metadata: The timestamp metadata was expired.

This means the Target to update to has expired.

**Solution:** Create a new Target for the same tag.

.. code-block::

   Failed to update Image repo metadata: Failed to fetch role timestamp in image repository.

> This could mean that there is no Target available to update to. If this is a production device, it could mean that there are no :ref:`ref-production-targets`/waves available for that tag.

Reviewer 4: I would suggest to check whether the metadata are available by the following two commands:

1. on device `curl -H "x-ats-tags: $DEVICE_TAG" https://$DG:8443/repo/targets.json --cert client.pem --key pkey.pem --cacert root.crt`

2. on a dev host: `curl -H "osf-token: $TOKEN" "https://api.foundries.io/ota/repo/$FACTORY/api/v1/user_repo/targets.json?tag=master&production=1"`

Author: I replaced the dev host command with: fioctl targets list --by-tag <tag> --production

**Solution:** :ref:`Create a wave <ref-rm-wave>` for the wanted tag.

* **Configuration file wrong or corrupted**

.. code-block::

   Configuration file wrong or corrupted
   warning: Failed resetting bootcount

> This means that the device cannot access the U-Boot environment.

Author: I think it's only for u-boot environment where it reads stuff from /mnt/uboot.env afaiu.

> This means that the device cannot access the U-Boot environment.

Author: Maybe @rsalveti can comment.

> This means that the device cannot access the U-Boot environment.

Reviewer 3: I think we only see this message when u-boot env is not mounted correctly.

**Solution:** Check if ``fstab`` is properly set.

* **KeyId is not valid**

.. code-block::

   KeyId xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx is not valid to sign for this role (root).

> This is not an actual error. It only indicates that the TUF root key has been rotated. It can be shown more than once in the ``aktualizr-lite`` logs depending on how many times the TUF root key has been rotated.

Reviewer 2: ```suggestion
This is not an actual error. It only indicates that the TUF root key has been rotated. It can be shown more than once in the ``aktualizr-lite`` logs depending on how often the TUF root key has been rotated.
```
